noDeltaStream

vladsud opened the pull request:
Breaking out PR #7422 into smaller chunks.

This change modified pre-conditions RE closed state & presence of handler in DeltaManager.
But most important, it stops fetching of ops when handler is not installed, as there is not enough information at such moment to fetch ops. That's because op handler missing means we do not know sequence number of snapshot we loaded from, and thus can't really load ops yet - we do not know where to start.

This leads to discovery of issue #7426 that would need to be addressed properly in some way.
I'm not solving it here, but rather patching UTs to pass given the tools I have.
Right follow up to the problem described in this issue is required.

vladsud committed:
noDeltaStream

vladsud committed:
assert message

vladsud committed:
Merge branch 'main' of https://github.com/microsoft/FluidFramework into noDeltaStream

# Conflicts:
#	packages/loader/container-loader/src/deltaManager.ts

jatgarg commented on packages/loader/container-loader/src/deltaManager.ts:
```diff
@@ -586,9 +586,14 @@ export class DeltaManager
-        // Note that might already got connected to delta stream by now.
-        // If we did, then we proactively fetch ops at the end of setupNewSuccessfulConnection to ensure
-        if (this.connection === undefined) {
+        // There should be never pending fetch!
```
nit: "never" ->  "no"

vladsud committed:
PR feedback / comment

vladsud committed:
Reorginize code around attachopHandler / prefetchOps, as newly added assert points to another problem in this code.
attachopHandler would start processing pending ops, which would completely screw prefetching call that would follow, as there would be already some fetchMissingOps() call in flight.
So prefetching should come first, and we also want to assert at the end of this process that either there is pending fetch or there are no pending ops (as there can't be due to matrix involved).
So reorganize code for all this logic to be one single funciton that can make proper decisions and assertions
